services/users/src/routes/company.ts

```diff
@@ -50,35 +56,79 @@ companyRoutes.route("/:id").put(
   })
 );
 
+// get company based on employee id provided
+companyRoutes.route("/employees/:employeeId").get(
+  checkAbility("read", "Company"),
+  asyncHandler(async (req, res) => {
+    const company = await CompanyModel.findOne({ employees: req.params.employeeId });
+
+    if (!company) {
+      throw new BadRequestError("Company not found or you do not have permission.");
+    }
+
+    return res.status(200).send(company);
+  })
+);
+
 companyRoutes.route("/:id/employees/add").post(
   checkAbility("update", "Company"),
   asyncHandler(async (req, res) => {
-    const company = await CompanyModel.findById(req.params.id).accessibleBy(req.ability);
+    const company = await CompanyModel.findById(req.params.id);
 
     if (!company) {
       throw new BadRequestError("Company not found or you do not have permission.");
     }
 
-    const emails = req.body.employees.split(",");
-    const employeesToAdd: string[] = [];
+    const employeesToAdd = req.body.employees.split(",");
     const uniqueEmployees: string[] = company.employees;
 
-    emails.forEach(async (email: string) => {
-      const user = await getAuth().getUserByEmail(email);
-      if (!uniqueEmployees.includes(user.uid)) {
-        uniqueEmployees.push(user.uid);
-      }
```

Comment:
So I'm pretty sure you still have to keep this code to retrieve the user email from firebase, because you have to lookup and get their uid by their email. Otherwise, you won't know it.

Reply:
I was going to start saving the employee's userIds rather than their emails. Do you see any reasons that might not work?

Reply:
Yeah so the code I had before saved the userId im pretty sure? What i did here with this code `getAuth().getUserByEmail(email);` is that the person makes a request with the email of the employee they want to add, and then i use firebase authentication to find the user id for that user. Then, this is the userid that i save to the database.

I think after u changed the code, you have to send someones userid directly, which is not something people on the frontend would know.

Reply:
Aah I see! I'll revert that back